gridstatus/ieso.py

```diff
@@ -942,6 +942,62 @@ def get_resource_adequacy_report(
         logger.debug(f"DataFrame Shape: {df.shape}")
         return df.sort_values(["Interval Start", "Publish Time", "Last Modified"])
 
+    def get_resource_adequacy_report_by_last_modified(
+        self,
+        last_modified: str | datetime.date | datetime.datetime,
+        vintage: Literal["all", "latest"] = "latest",
+    ) -> pd.DataFrame:
+        """Retrieve and parse Resource Adequacy Reports modified after last_modified time.
+        This method bypasses date iteration and gets all files across all dates.
+        This is useful for ETL systems that want to get all new files at once.
+
+        Args:
+            last_modified: The last modified time after which to get report(s)
+            vintage: The version of the report to get
+
+        Returns:
+            pd.DataFrame: The Resource Adequacy Report df with all files modified after last_modified
+        """
+        if last_modified:
+            last_modified = utils._handle_date(last_modified, tz=self.default_timezone)
+
+        if vintage == "all":
+            json_data_with_times = (
+                self._get_all_resource_adequacy_jsons_by_last_modified(
+                    last_modified,
+                )
+            )
+            dfs = []
+            for json_data, file_last_modified in json_data_with_times:
+                try:
+                    df = self._parse_resource_adequacy_report(json_data)
+                    df["Last Modified"] = file_last_modified
+                    dfs.append(df)
+                except Exception as e:
+                    logger.warning(
+                        f"Failed to parse resource adequacy report: {str(e)}",
+                    )
+                    continue
+
+            if not dfs:
+                return pd.DataFrame()
+
+            df = pd.concat(dfs, ignore_index=True) if len(dfs) > 1 else dfs[0]
+        else:
+            raise ValueError("cross_date_fetch only supported with vintage='all'")
```

Comment:
If this method only works with `vintage="all"`, then should we remove the vintage parameter from the method?